src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/CSharpTagHelperCodeRenderer.cs

```diff
@@ -341,27 +347,34 @@ private void RenderUnboundHTMLAttributes(IEnumerable<KeyValuePair<string, Chunk>
                     continue;
                 }
 
+                var addHtmlAttributeMethodName = attributeValue == null ?
+                    _tagHelperContext.ExecutionContextAddMinimizedHtmlAttributeMethodName :
+                    _tagHelperContext.ExecutionContextAddHtmlAttributeMethodName;
```

Comment:
Remove this ternary and move the similar-but-not-similar-enough `WriteStartInstanceMethodInvocation()` calls into the following `if` block.  Much easier to understand with fewer checks of `attributeValue == null` and all of the method invocation together.